napi-inl.h

```diff
@@ -11,19 +11,24 @@
 
 namespace Napi {
 
-// Adapt the NODE_MODULE registration function to NAPI:
+// Adapt the NAPI_MODULE registration function:
 //  - Wrap the arguments in NAPI wrappers.
 //  - Catch any NAPI errors that might be thrown.
-#define NAPI_MODULE(modname, regfunc)                                                 \
-  void __napi_ ## regfunc(napi_env env, napi_value exports, napi_value module) {      \
-    try {                                                                             \
-      regfunc(Napi::Env(env), Napi::Object(env, exports), Napi::Object(env, module)); \
-    }                                                                                 \
-    catch (const Napi::Error&) {                                                      \
-      assert(false); /* Uncaught error in native module registration. */              \
-    }                                                                                 \
-  }                                                                                   \
-  NODE_MODULE_ABI(modname, __napi_ ## regfunc);
+#define NODE_API_MODULE(modname, regfunc)                                \
+  void __napi_ ## regfunc(napi_env env,                                  \
+                          napi_value exports,                            \
+                          napi_value module,                             \
+                          void* priv) {                                  \
+    try {                                                                \
+      regfunc(Napi::Env(env),                                            \
+              Napi::Object(env, exports),                                \
+              Napi::Object(env, module));                                \
+    }                                                                    \
+    catch (const Napi::Error&) {                                         \
+      assert(false); /* Uncaught error in native module registration. */ \
```

Comment:
Good question. Now that I think about it, the correct thing to do here would be to throw a JavaScript exception. (If there is not already a pending exception.) [That is exactly what the callback wrapper does](https://github.com/nodejs/node-api/blob/master/napi-inl.h#L976-L981), and it should work just as well here. I think I coded this `assert()` before I designed the error-to-exception mechanism.

I'll update this PR.